src/Version.h

```diff
@@ -17,26 +18,81 @@
 
 namespace Version {
 
-using exact_width_mumble_raw_version_t = std::uint32_t;
-using mumble_raw_version_t             = unsigned int;
+enum class CompareMode { AtLeast, LessThan };
 
-static_assert(sizeof(exact_width_mumble_raw_version_t) <= sizeof(mumble_raw_version_t),
-			  "Invalid scalar type used to represent Mumble versions");
+//
+// The mumble version format (v2) is a uint64:
+// reserved/unused  major   minor   patch
+// 0xFFFF           0xFFFF  0xFFFF  0xFFFF
+// (big-endian)
+//
+
+using mumble_raw_version_t = std::uint64_t;
```

Comment:
```suggestion
using raw_t = std::uint64_t;
```
This thing does live in the `Version` namespace after all.
Maybe something like `bitmasked_t` might even be a better name?

(I know this is unrelated to the actual issue we're trying to fix but since we're touching all this code anyway... :shrug: )

Reply:
or ``blob_t`` as in libmumble, but in my eyes this is not a binary large object...

Reply:
> (I know this is unrelated to the actual issue we're trying to fix but since we're touching all this code anyway... shrug )


Mumble dev team: "Let's not rewrite Mumble. Refactor as we go..."
Me, trying to fix a thing or two: ""
Mumble dev team: "While you are at it..."

Reply:
> but in my eyes this is not a binary large object...

Agreed.

> Mumble dev team: "Let's not rewrite Mumble. Refactor as we go..."
Me, trying to fix a thing or two: ""
Mumble dev team: "While you are at it..."

:joy: :see_no_evil:

Reply:
@Krzmbrzl What do you think of ``Version::full_t`` as a name for the full version object? (Similar to ``component_t``)

Reply:
Sounds good :+1: